hubblestack/loader.py

```diff
@@ -0,0 +1,1292 @@
+                    sys.exit(salt.defaults.exitcodes.EX_OK)
+            log.error(
+                'Failed to import %s %s as the module called exit()\n',
+                self.tag, name, exc_info=True
+            )
+            self.missing_modules[name] = error
+            return False
+        finally:
+            sys.path.remove(fpath_dirname)
+
+        if hasattr(mod, '__opts__'):
+            mod.__opts__.update(self.opts)
+        else:
+            mod.__opts__ = self.opts
+
+        # pack whatever other globals we were asked to
+        for p_name, p_value in six.iteritems(self.pack):
+            setattr(mod, p_name, p_value)
+
+        module_name = mod.__name__.rsplit('.', 1)[-1]
+
+        # Call a module's initialization method if it exists
+        module_init = getattr(mod, '__init__', None)
+        if inspect.isfunction(module_init):
+            try:
+                module_init(self.opts)
+            except TypeError as e:
+                log.error(e)
+            except Exception:
+                err_string = '__init__ failed'
+                log.debug(
+                    'Error loading %s.%s: %s',
+                    self.tag, module_name, err_string, exc_info=True
+                )
+                self.missing_modules[module_name] = err_string
+                self.missing_modules[name] = err_string
+                return False
+
+        # if virtual modules are enabled, we need to look for the
+        # __virtual__() function inside that module and run it.
+        if self.virtual_enable:
+            virtual_funcs_to_process = ['__virtual__'] + self.virtual_funcs
+            for virtual_func in virtual_funcs_to_process:
+                virtual_ret, module_name, virtual_err, virtual_aliases = \
+                    self._process_virtual(mod, module_name, virtual_func)
+                if virtual_err is not None:
+                    log.trace(
+                        'Error loading %s.%s: %s',
+                        self.tag, module_name, virtual_err
+                    )
+
+                # if _process_virtual returned a non-True value then we are
+                # supposed to not process this module
+                if virtual_ret is not True and module_name not in self.missing_modules:
+                    # If a module has information about why it could not be loaded, record it
+                    self.missing_modules[module_name] = virtual_err
+                    self.missing_modules[name] = virtual_err
+                    return False
+        else:
+            virtual_aliases = ()
+
+        # If this is a proxy minion then MOST modules cannot work. Therefore, require that
+        # any module that does work with salt-proxy-minion define __proxyenabled__ as a list
+        # containing the names of the proxy types that the module supports.
+        #
+        # Render modules and state modules are OK though
+        if 'proxy' in self.opts:
```

Comment:
It is for proxy minions

Reply:
FYI, I figure at least some of the mods will expect proxy arguments and the global `__proxy__` to exist; so I didn't strip it out of args lists and still provide the `__proxy__` global to modules at load. 

You're right about this one for sure though. removed.